### ISSUE TITLE
[Feature] Allow managing enrich policies

### ISSUE THREAD

**Issue by jonjitsu** [elastic/terraform-provider-elasticstack#282]
**Describe the resource you would like to have implemented.**
This ticket is about increasing the API coverage of this provider to include the enrich policy API.

**Describe the solution you'd like**

```terraform
resource "elasticstack_elasticsearch_enrich_policy" "my-policy" {
  # required
  policy_type = "match"
  indices = []
  match_field = ""
  enrich_fields = []
  # optional
  query = ""
  # maybe 
  execute = false
}
```
By default it could execute any created policies, but through an optional `execute` flag can be configured to not execute.

**Describe alternatives you've considered**
I've used Mastercard/restapi to manage enrich policies but it's clunky since all updates need the `terraform -replace=resource...` mechanism.

I can send a PR for this with some guidance.

**Comment by tobio** [comment]
> I can send a PR for this with some guidance.

👍 that would be great! I'm happy to help with any reviews or guidance on how the provider is put together. The existing Elasticsearch resources are all within [this](https://github.com/elastic/terraform-provider-elasticstack/tree/main/internal/elasticsearch) package. The [data stream](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/elasticsearch/index/data_stream.go) resource might be a good example to look at. 

If you do look at this, feel free to ping me with any questions you have :)

**Comment by jonjitsu** [comment]
A quick update, I started on this yesterday, so it's partially done (my golang is a bit rusty).

For PRs how big do we make them? For the Enrich API, in order to have complete coverage, there would be one resource and two data sources (one for looking up enrich policies and one for the stats endpoint). Should all this be in a single PR or split out into three, I lean towards smaller units so three PRs.

> If you do look at this, feel free to ping me with any questions you have :)

Where to ping you? I'm in the elastic slack workspace.

**Comment by tobio** [comment]
> Where to ping you?

Here is fine.

> For PRs how big do we make them?

+1 to splitting this across 2/3 PRs. Definitely worth keeping resources and data sources independent, there's been cases where it's made sense to do two data sources in a single PR. It doesn't sound like that's the case here, but not a problem if you end up combining them.

**Comment by Kushmaro** [comment]
Closing, this has been implemented in v0.6
